gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsException.java

```diff
@@ -31,5 +33,21 @@ public DnsException(IOException exception) {
     super(exception, true);
   }
 
+  public DnsException(int code, String message) {
+    super(code, message, null, true);
+  }
+
+  /**
+   * Translate RetryHelperException to the DnsException that caused the error. This method will
+   * always throw an exception.
+   *
+   * @throws DnsException when {@code ex} was caused by a {@code DnsException}
+   * @throws RetryInterruptedException when {@code ex} is a {@code RetryInterruptedException}
+   */
+  static DnsException translateAndThrow(RetryHelperException ex) {
+    BaseServiceException.translateAndPropagateIfPossible(ex);
+    throw new DnsException(UNKNOWN_CODE, ex.getMessage());
+  }
+
```

Comment:
regarding the comment bellow, though doc not in par with other services I think there is something that we can start using (and refer to in the javadoc) -> https://cloud.google.com/dns/troubleshooting